UI/js/scriptForView.js

```diff
@@ -370,51 +549,258 @@ let offers = new Model([
+                <button class="sign-in-button">Sign in</button>
+            </form>
+        `
+    }
+
+    function loadInfoLogInOut() {
+        if (view.getIsLogIn()) {
+            view.setIsLogIn(false);
+            view.setIsVendor(false);
+            view.setUsername('Guest');
+            view.refreshPage();
+        } else {
+            view.setPage(document.querySelector(".page"));
+            document.querySelector(".page").remove();
+            let newPage = document.createElement("page");
+            newPage.className = "log-in-page";
+            newPage.innerHTML = getLogInPage();
+            document.querySelector('.header').after(newPage);
+        }
+    }
+
+    function handleLogInOutButton() {
+        loadInfoLogInOut();
+        if (document.querySelector(".log-in-page") != null) {
+            document.forms[0].addEventListener('submit', handleSignInButton);
+        }
+    }
+
+    function handleSignInButton(event) {
+        event.preventDefault();
+        let form = event.target;
+        if (form.elements.user.value !== '' && form.elements.password.value !== '') {
+            view.setUsername(form.elements.user.value);
+            view.setIsLogIn(true);
+            view.setIsVendor(true);
+            form.removeEventListener('submit', handleSignInButton);
+            document.querySelector(".log-in-page").remove();
+            document.querySelector('.header').after(view.getViewPage());
+            view.refreshPage();
+        }
+    }
+
+    let addNewOfferButton = document.querySelector(".add-new-offer-button");
+    addNewOfferButton.addEventListener('click', handleAddNewOfferButton);
+
+    function getAddEditOfferPage(message){
+        return `
+            <label class="add-new-offer-message">${message}</label>
+            <form class='add-new-offer-form'>
+               <label class="label-class">Service: <input type="text" class="label" name ="label" placeholder="Name of your service"></label>
+               <label class="description-class">Short <br> description: <input type="text" class="description" name ="description" placeholder="A short description"></label>
+               <label class="discount-class">Discount: <input type="text" class="discount" name ="discount" placeholder="Percentage"></label>
+               <label class="date-class">Offer expires: <input type="date" class="date-to" name ="dateTo"></label>
+               <label class="website-class">Website: <input type="text" class="link" name ="link" placeholder="Link to your website"></label>
+               <label class="photo-class">Photo link: <input type="text" class="photo-link" name ="photoLink" placeholder="Link to the photo"></label>
+               <label class="hashtags-class">Hashtags: <input type="text" class="hashtags" name ="hashtags" placeholder="At least one hashtag"></label>
+               <button class="save-new-offer-button" type="submit">Save</button>
+            </form>
+        `
```

Comment:
Controller code should only add event listeners and call corresponding methods of view class, here you're mixing view and controller logic.